swarm/tools/lint_routing_fields.py

```
@@ -337,9 +337,26 @@ def check_file(
+    while stack:
+        current = stack.pop()
+        try:
+            with os.scandir(current) as it:
+                for entry in it:
+                    try:
+                        if entry.is_dir(follow_symlinks=False):
+                            # Skip .git and common excluded dirs
+                            if not entry.name.startswith(".") and entry.name not in ["__pycache__", "node_modules"]:
+                                stack.append(entry.path)
+                        elif entry.is_file(follow_symlinks=False):
+                            if any(entry.name.endswith(ext) for ext in CHECK_EXTENSIONS):
```

Ruff (0.15.10)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)